grcs/0020-knowledge-graph.md

```diff
@@ -199,95 +211,79 @@ Every entity has an implicit Types relation and can have zero, one or many types
-| From entity     | Attribute | RERshk4JoYoMC17r1qAo9J |
-| To entity       | Attribute | Qx8dASiTNsxxP3rJbd4Lzd |
-| Index           | Attribute | WNopXUYxsSsE51gkJGWghe |
+A checkbox can either be `true` or `false`. It's a boolean field named for normal people. The string encoding of the checkbox field must be "1" for `true` and "0" for `false`. Checkbox values can be rendered with alternate UI components such as switches and toggles.
 
-From entity, To entity and Index are required.
+#### 5.4 URL
 
-#### 8.1 Relation Types
+A URL value type is technically a URI that starts with the protocol identifier followed by a :// and the resource. The initial supported protocols are: graph, ipfs, ar, https.
 
-A Relation Type is a definition of a relation similar to an attribute but specifically for one to one, one to many, and many to many relations. A Relation Type can have a name like `Team member`, `City` or a more graph style name like `Born in`. A relation must have exactly one Relation Type.
+Examples:
+* "ipfs://QmVGF2e9WqF8g39W81eCQxecz7Bdi5o2DFSJ5BQxwfveYB"
+* "ht<span>tps://</span>geobrowser.io"
 
-| Name          | Type      | ID                     |
-| ------------- |-----------|----------------------- |
-| Relation Type | Type      | 3WxYoAVreE4qFhkDUs5J3q |
+#### 5.5 Time
 
-A Relation Type is also of type `Type` and has the following attributes:
+Time is represented as an ISO-8601 format string. A time value can be used to describe both date and time as well as durations. Clients should include timezones when creating times. In rare exceptions, if the timezone is omitted, the time is interpreted as being in the consumer's local time. Example: `2024-11-03T11:15:45.000Z`
 
-| Name                     | Type              | Description                                                                    |
-| ------------------------ |-------------------| ------------------------------------------------------------------------------ |
-| Relation value types     | Relation \<Type>  | A list of types that the relation To entity may have. Used as a hint           |
-| One max                  | Checkbox          | Whether to limit the number of relations from a given entity to no more than 1 |
+Options: `format`
 
-| Name                     | Type          | ID                     |
-| ------------------------ |---------------|----------------------- |
-| Relation value types     | Relation Type | SeDR2b1bdnb6tj2qmdcFdg |
-| One max                  | Checkbox      | K1M9AEyda5oNB2WS38bDDW |
+A format string using the Unicode Technical Standard #35.
+
+Examples:
+* "MMMM do, yyyy" - July 4th, 2024
+* "M/d/yyyy" - 7/4/2024
+* "h:mma" - 4:45pm
+* "EEEE h:mma" - Friday 5:00pm
+
+#### 5.6 Point
+
+A point is a location specified with x and y coordinates string encoded. Points can be used for geo locations or other cartesian coordinate systems. Clients may choose how many dimensions to support. Example: `"12.554564, 11.323474"`
```

Comment:
Byron: Should we also support Z dimensions?

It says that clients may choose how many dimensions to support, so we should be able to support multidimensional points already. What we can't do yet is support things like radial coordinate systems. We should add a coordinate system property to the Property that has type point to specific the coordinate system. I'm not sure though if we should expect that an indexer can support different coordinate systems outside of cartesian. We could start by allowing different coordinate systems which are cartesian (2 or 3 dimensions) for different environments / physical spaces.

Reply:
Currently implementing Point in Hypergraph and I'm wondering if it would be nicer for users to have more explicit types e.g.

```
type Point2D = [number, number];
type Point3D = [number, number, number];
type Point = number[];
```